lib/puppet/type/ini_setting.rb

```diff
@@ -5,7 +5,19 @@
   desc 'ini_settings is used to manage a single setting in an INI file'
   ensurable do
     desc 'Ensurable method handles modeling creation. It creates an ensure property'
-    defaultvalues
+    newvalue(:present) do
+      provider.create
+    end
+    newvalue(:absent) do
+      provider.destroy
+    end
+    def insync?(current)
+      if @resource[:refreshonly]
+        true
+      else
+        current == should
```

Comment:
I am unsure what this line means, can you please elaborate?
Otherwise it looks like a great PR! 👍

Reply:
Never mind apologies, I expanded the file and this term is used throughout the code base.